docs/rfcs/1003-mandatory-human-handle.md

```diff
@@ -0,0 +1,235 @@
+- The user id accepts unicode characters (encoded as UTF-8) while
+  [email address is more messy about it](https://en.wikipedia.org/wiki/International_email)
+- The email format is divided into `<username>@<domain>`
+
+For the domain part we use the [standard `.invalid` tld](https://en.wikipedia.org/wiki/.invalid).
+On top of that we use `redacted.invalid` as domain, considering the only non-legacy reason
+for not having a human handle set is because the certificate is a redacted one (in practice
+there is very few organization old enough to need legacy support, hence not having a
+specific way to handle not-redacted-but-also-no-human-handle-field is not worth it).
+
+For the username part we cannot use the user id field as-is, but instead apply to it
+the `uncaseify` algorithm that encode the case.
+
+`uncaseify` is as follow:
+
+1 - replace any `_` by `__`
+2 - replace any uppercase character by underscore followed by the lowercase version of the character
+
+```python
+def uncaseify(txt: str) -> str:
+    out = ""
+    for c in txt:
+        if c.isupper():
+            out += "_"
+            out += c
+        elif c == "_":
+            out += "__"
+        else:
+            out += c
+    return out
+
+assert uncaseify("a") == "a"
+assert uncaseify("_a_") == "__a__"
+assert uncaseify("aAa") == "a_aa"
+assert uncaseify("AA") == "_a_a"
+assert uncaseify("_A_") == "___a__"
+```
+
+`UserID` type has a hard limit of 32 bytes, while the email email address's limit
+is 255 bytes (including the domain part and `@`).
+The worst case for `uncaseify` is a user id only composed of uppercase (or underscores),
+in which case it size is doubled so 64 bytes.
+Considering domain is `redacted.invalid` (16 bytes long) and the `@`, there is
+238 bytes to accommodate the uncaseified user id (so far more than what is needed).
+
+Finally considering the unicode, the good news is there is nothing to do: our email
+format already accept international email address (so a regular user can already use
+something like `关羽@蜀.汉`).
+
+### 4 - `redacted.invalid` not allowed for real human handle
+
+Let's consider the following scenario:
+
+1 - User 1 (user id = "alice") is created with an old Parsec version, hence it has no human handle
+2 - User 2 (user id = "bob") is created with human handle email = "alice@redacted.invalid"
+
+Now we end up with Parsec displaying the same human handle email for both user 1 and 2 :(
+The easy solution for this is simply to consider `redacted.invalid` as a reserved domain that
+is not allowed for real human handle.
+
+Typically, `HumanHandle` method would look like:
+
+```rust
+impl HumanHandle {
+    pub fn new(email: &str, label: &str) -> Result<Self, &'static str> {
+      // Create regular human handle from the provided email & label
+      // here we ensure email doesn't use `redacted.invalid` domain
+    }
+
+    pub fn from_redacted_user(user_id: &UserID) -> Self {
```

Comment:
```suggestion
    pub fn from_redacted_user(user_id: &UserID) -> Result<Self, &'static str> {
```

Will likely use `HumanHandle::new` since either way we will need to validate the email

Reply:
We can always create a valid `HumanHandle` from a `UserID`, so no need to return an error